ipatests/test_integration/test_otp.py

```diff
@@ -398,6 +460,114 @@ def test_2fa_disable_single_prompt(self):
             self.master.run_command(['semanage', 'login', '-D'])
             sssd_conf_backup.restore()
 
+    def test_2fa_only_with_password(self):
+        """Test ssh with 2FA only with the password(first factor) when
+        user-auth-type is opt and password.
+
+        Test for :  https://github.com/SSSD/sssd/pull/7500
+
+        Add the IPA user and user-auth-type set to opt and password.
+        Authenticate the user only with password, just press enter
+        at `Second factor`
+        """
+        master = self.master
+        USER3 = 'sshuser3'
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        first_prompt = 'Enter first factor:'
+        second_prompt = 'Enter second factor:'
+        add_contents = textwrap.dedent('''
+            [prompting/2fa/sshd]
+            single_prompt = False
+            first_prompt = {0}
+            second_prompt = {1}
+            ''').format(first_prompt, second_prompt)
+        set_sssd_conf(master, add_contents)
+        tasks.create_active_user(master, USER3, PASSWORD)
+        tasks.kinit_admin(master)
+        master.run_command(['ipa', 'user-mod', USER3, '--user-auth-type=otp',
+                            '--user-auth-type=password'])
+        try:
+            otpuid, totp = add_otptoken(master, USER3, otptype='totp')
+            master.run_command(['ipa', 'otptoken-show', otpuid])
+            totp.generate(int(time.time())).decode('ascii')
+            otpvalue = "\n"
+            tasks.clear_sssd_cache(self.master)
+            github_ticket = "https://github.com/SSSD/sssd/pull/7500"
+            sssd_version = tasks.get_sssd_version(master)
+            fedora_fail = (
+                osinfo.id == 'fedora'
+                and sssd_version < tasks.parse_version("2.10.0")
```

Comment:
I checked on f39 also and the issue was introduced in sssd 2.9.5 (2.9.4 doesn't have the regression and our CI also runs on f39). In order to have an accurate condition, can you replace with:
```
            fedora_fail = (
                osinfo.id == 'fedora'
                and sssd_version == tasks.parse_version("2.9.5")
```